tests/std/tests/P0896R4_views_common/test.cpp

```diff
@@ -0,0 +1,383 @@
+            static_assert(noexcept(as_const(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | pipeline), RC>);
+            static_assert(noexcept(as_const(rng) | pipeline) == is_noexcept);
+        }
+
+        // ... with rvalue argument
+        static_assert(CanViewCommon<remove_reference_t<Rng>> == is_view || enable_borrowed_range<remove_cvref_t<Rng>>);
+        if constexpr (is_view) {
+            constexpr bool is_noexcept = is_nothrow_move_constructible_v<V>;
+            static_assert(same_as<decltype(views::common(move(rng))), R>);
+            static_assert(noexcept(views::common(move(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common), R>);
+            static_assert(noexcept(move(rng) | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common | views::common | views::common), R>);
+            static_assert(noexcept(move(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | pipeline), R>);
+            static_assert(noexcept(move(rng) | pipeline) == is_noexcept);
+        } else if constexpr (enable_borrowed_range<Rng>) {
+            using S                    = decltype(ranges::subrange{declval<Rng>()});
+            using RS                   = common_view<S>;
+            constexpr bool is_noexcept = noexcept(S{declval<Rng>()});
+
+            static_assert(same_as<decltype(views::common(move(rng))), RS>);
+            static_assert(noexcept(views::common(move(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common), RS>);
+            static_assert(noexcept(move(rng) | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common | views::common | views::common), RS>);
+            static_assert(noexcept(move(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | pipeline), RS>);
+            static_assert(noexcept(move(rng) | pipeline) == is_noexcept);
+        }
+
+        // ... with const rvalue argument
+        static_assert(CanViewCommon<const remove_reference_t<Rng>> == (is_view && copyable<V>)
+                      || (!is_view && enable_borrowed_range<remove_cvref_t<Rng>>) );
+        if constexpr (is_view && copyable<V>) {
+            constexpr bool is_noexcept = is_nothrow_copy_constructible_v<V>;
+
+            static_assert(same_as<decltype(views::common(move(as_const(rng)))), R>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_nothrow_copy_constructible_v<R>);
+
+            static_assert(same_as<decltype(move(as_const(rng)) | views::common), R>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_nothrow_copy_constructible_v<R>);

```

Comment:
> @CaseyCarter thinks that this may be a bug copy-pasted from other ranges tests.

@CaseyCarter was right. There's also a `move` missing on 171. These two occurrences were missing from _every_ range adaptor closure (pipeable thing) test. I submitted #1393 to correct occurrences that have been merged, and submitted code review comments to correct those that haven't yet merged.

Reply:
found 2 more to be missing and added the move